flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -133,13 +139,14 @@ public TaskManagerRunner(Configuration configuration, ResourceID resourceId) thr
 			HighAvailabilityServicesUtils.AddressResolution.TRY_ADDRESS_RESOLUTION);
 
 		rpcService = createRpcService(configuration, highAvailabilityServices);
+		metricQueryRpcService = createMetricQueryRpcService(configuration, highAvailabilityServices);
```

Comment:
Same as in the `ClusterEntrypoint`. We only need an `ActorSystem` here.

Reply:
We should also shut down this resource in the `shutDownServices` method.

Reply:
I will do that.